std/io/bufio.ts

```diff
@@ -571,7 +571,7 @@ export class BufWriterSync extends AbstractBufBase implements WriterSync {
     let numBytesWritten = 0;
     try {
       numBytesWritten = this.writer.writeSync(
-        this.buf.subarray(0, this.usedBufferBytes)
+        this.buf.slice(0, this.usedBufferBytes)
```

Comment:
Test only demonstrates the async case... So this change is untested

Reply:
I'll submit a test, we don't have a `StringWriter.writeSync` method though. I can implement it in this PR too or submit a new one with that addition if we do want a synchronous method on `StringWriter`.

Otherwise I'll write a test without using `StringWriter`.